src/main/java/com/nirima/jenkins/plugins/docker/DockerCloud.java

```diff
@@ -28,6 +28,7 @@
 import jenkins.authentication.tokens.api.AuthenticationTokens;
 import jenkins.model.Jenkins;
 import org.apache.commons.codec.binary.Base64;
+import org.jenkinsci.plugins.cloudstats.ProvisioningActivity;
```

Comment:
In the PR description, you state that the dependency on the cloudstats plugin is optional.
Here, we've got an import that'll cause this class (which is core to this plugin) to fail to classload if the cloudstats plugin is not available.
i.e. either the description needs work, or this code does.

Reply:
The JLS requires classloading to be deferred until the class is referenced by another class (though most classloaders are lazy, and defer loading even longer until a need for class initialization occurs). 

In this proposal, there are no *direct* references to `com.nirima.jenkins.plugins.docker.cloudstat.*` package from the entry point - the only reference made is indirect (reflection) loading in [`src/main/java/com/nirima/jenkins/plugins/docker/DockerNodeFactory.java`](https://github.com/jenkinsci/docker-plugin/compare/master...theshadow27:cloud-stats-support#diff-5d3d42111eb185179cd4f284f9f18318R26), on line 26:
```
return (DockerNodeFactory)Class.forName("com.nirima.jenkins.plugins.docker.cloudstat.CloudStatsFactory").newInstance();
```
This is the first (and only) time that the `CloudStatsFactory` class is initialized. You are correct that this class references the optional dependencies which may be missing, and will fail to class load if they are not there - but the `try` block around this line ensures that a default stub implementation is used instead in that case.